### ISSUE TITLE
Use of undeclared Var, 0.24.0 only (:test related)

### ISSUE THREAD

**Issue by eraserhd** [oracle-samples/clara-rules#513]
This might be ClojureScript specific, as this project is just ClojureScript.

```clojure
(defrecord Fact [e a v])
(defn asserted [e a v]
  (assoc (->Fact e a v) :persistent? true))

(clara/defquery persistent-facts []
  [?fact <- Fact (= e ?e) (= a ?a) (= v ?v)]
  [:test (:persistent? ?fact)])
```

```
 Use of undeclared Var cadro.session/?fact
```

This also warns with 0.24.0:

```clojure
(clara/defquery persistent-facts []
  [?fact <- Fact (= e ?e) (= a ?a) (= v ?v)]
  [:test (= ?e 2)])
```

```
 Use of undeclared Var cadro.session/?e
```

As does this:

```clojure
(clara/defrule only-one-reference
  [?refcount <- (acc/count) :from [Fact (= a ::reference?)]]
  [:test (<= 2 ?refcount)]
  =>
  (clara/insert! (->InvariantError "more than one reference point in session")))
```

None of these warn for 0.22.1.

At least the last rule appears to work in spite of warning.

**Comment by eraserhd** [comment]
First broken commit: 759ab4045e76bbe54c1014dfea2cc56605d560ac

**Comment by eraserhd** [comment]
The solution appears to be this, but making sure it is right:

```diff
diff --git a/src/main/clojure/clara/macros.clj b/src/main/clojure/clara/macros.clj
index 81f4234..9b76c51 100644
--- a/src/main/clojure/clara/macros.clj
+++ b/src/main/clojure/clara/macros.clj
@@ -132,7 +132,7 @@
         `(eng/->TestNode
           ~id
           ~(:env beta-node)
-          ~(:constraints condition)
+          '~(:constraints condition)
           ~(com/compile-test id (:constraints condition) (:env beta-node))
           ~(gen-beta-network child-ids beta-graph all-bindings))
```
